client/src/pages/ContestDetails.js

```diff
@@ -143,21 +161,40 @@ export default function ContestDetails(props) {
         setOpenDesignDialog(false);
     }
 
+    const handlePaymentDialogClose = () => {
+        setOpenPaymentDialog(false)
+    }
+
     const handleWinnerChange = e => {
         setWinningSubmission(parseInt(e.target.value))
     }
 
-    const onWinnerSubmit = e => {
-        setContestWinner(contestId, winningSubmission, data => {
-            console.log('Winner Successfully Declared')
-        }, error => {
-            if (error instanceof RequestError && error.status === 400) {
-                console.log(error.body)
-            } else {
-                console.log("unexpected error")
+    const onWinnerSubmit = async(e) => {
+        var intentSecret = null
```

Comment:
avoid using `var` as much as possible. do you need this variable to be at this level? can you call it and initilize it on line 176?